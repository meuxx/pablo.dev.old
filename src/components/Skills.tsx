import React from 'react'
import { Section, SectionHeader, SkillSetWrapper, SkillSetTitle } from '../styles/styles'

interface SkillSetInterface {
  name: string
  skills: string[]
}

const skillsData: SkillSetInterface[] = [
  {
    name: 'JavaScript',
    skills: ['ES6', 'NodeJS', 'TypeScript', 'jQuery', 'Express', 'React', 'Gatsby'],
  },
  {
    name: 'PHP',
    skills: ['Symfony', 'Laravel', 'Composer', 'PDO', 'Doctrine'],
  },
  {
    name: 'Web',
    skills: ['CSS3', 'HTML5', 'Semantic', 'SPA', 'Rest/GraphQL APIs'],
  },
  {
    name: 'Data Storage',
    skills: ['MySQL', 'Redis', 'Memcache', 'Cassandra', 'MongoDB', 'DynamoDB'],
  },
  {
    name: 'Tools',
    skills: [
      'Git',
      'SVN',
      'Docker',
      'Vagrant',
      'Nginx',
      'Apache',
      'HAProxy',
      'Ansible',
      'Kibana',
      'NewRelic',
      'Pingdom',
      'Grafana',
      'Prometheus',
      'Zabbix',
      'GitlabCI',
    ],
  },
  {
    name: 'Cloud',
    skills: ['AWS', 'S3', 'Route53', 'DigitalOcean', 'Netlify'],
  },
  {
    name: 'Scripting',
    skills: ['Python', 'Bash', 'Batch'],
  },
  {
    name: 'Other',
    skills: ['Scrum', 'Kanban', 'PCI Compliance', 'SEO'],
  },
]

const Skills: React.FC = () => (
  <Section>
    <SectionHeader>
      <h2>Skills</h2>
    </SectionHeader>
    <SkillSetWrapper>
      {skillsData.map((skillSet, index) => (
        <div key={`skillSet-${index}`}>
          <SkillSetTitle>{skillSet.name}</SkillSetTitle>
          <span>{skillSet.skills.join(', ')}</span>
        </div>
      ))}
    </SkillSetWrapper>
  </Section>
)

export default Skills
