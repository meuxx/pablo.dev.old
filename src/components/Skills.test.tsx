import React from 'react'
import { render } from '@testing-library/react'
import Skills from './Skills'

describe('Skills', () => {
  it('renders without crashing', () => {
    const { container } = render(<Skills />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
