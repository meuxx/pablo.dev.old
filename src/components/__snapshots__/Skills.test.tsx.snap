// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders without crashing 1`] = `
.c0 {
  margin: 2em 0 0 0;
  padding: 2em 0 0 0;
}

.c0:first-child {
  margin-top: 0;
  padding-top: 0;
}

.c0 p {
  color: #b2b2b2;
  position: relative;
  margin: 0 0 2em 0;
}

.c0 p strong {
  color: #ffffff;
  font-weight: 400;
}

.c1 h1,
.c1 h2 {
  color: #ffffff;
  font-weight: 400;
}

.c1 h1 {
  margin: 0 0 1em;
  line-height: 1.5em;
}

.c1 h2 {
  margin: 0 0 1em;
  font-size: 1.5em;
  line-height: 1.5em;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  gap: 1em;
}

.c3 {
  margin: 0;
  color: #0B61A4;
}

@media screen and (max-width:736px) {
  .c0 {
    margin: 1.5em 0 0 0;
    padding: 1.5em 0 0 0;
  }
}

@media screen and (max-width:736px) {
  .c1 h2 {
    font-size: 1.2em;
  }
}

@media screen and (max-width:480px) {
  .c2 {
    grid-template-columns: none;
  }
}

<div>
  <section
    class="c0"
  >
    <header
      class="c1"
    >
      <h2>
        Skills
      </h2>
    </header>
    <div
      class="c2"
    >
      <div>
        <h3
          class="c3"
        >
          JavaScript
        </h3>
        <span>
          ES6, NodeJS, TypeScript, jQuery, Express, React, Gatsby
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          PHP
        </h3>
        <span>
          Symfony, Laravel, Composer, PDO, Doctrine
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          Web
        </h3>
        <span>
          CSS3, HTML5, Semantic, SPA, Rest/GraphQL APIs
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          Data Storage
        </h3>
        <span>
          MySQL, Redis, Memcache, Cassandra, MongoDB, DynamoDB
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          Tools
        </h3>
        <span>
          Git, SVN, Docker, Vagrant, Nginx, Apache, HAProxy, Ansible, Kibana, NewRelic, Pingdom, Grafana, Prometheus, Zabbix, GitlabCI
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          Cloud
        </h3>
        <span>
          AWS, S3, Route53, DigitalOcean, Netlify
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          Scripting
        </h3>
        <span>
          Python, Bash, Batch
        </span>
      </div>
      <div>
        <h3
          class="c3"
        >
          Other
        </h3>
        <span>
          Scrum, Kanban, PCI Compliance, SEO
        </span>
      </div>
    </div>
  </section>
</div>
`;
